### PR TITLE
Code review

### PR DESCRIPTION
**Pull request by StuartCowley**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,6 +6,7 @@
     "@testing-library/jest-dom": "^5.16.5",
     "@testing-library/react": "^13.4.0",
     "@testing-library/user-event": "^13.5.0",
+    "axios": "^1.4.0",
```
Comment (StuartCowley) [inline package.json:9]: Your tests are not running as things stand here, you need to modify the test script in here to this: 
```
"test": "react-scripts test --transformIgnorePatterns \"node_modules/(?!axios)/\"",
```
This is nothing youve done wrong, its down to axios itself. The errors been mentioned in the slack channel a few times, but thankfully its an easy fix

File: src/components/Search.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+  const handleInputChange = (e) => setValue(e.target.value);
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    setSearchResults(await getImages(value));
+  };
+
+  return (
+    <>
```
Comment (StuartCowley) [inline src/components/Search.js:16]: You could remove the fragment here and just use the `form` as the wrapping element

File: src/components/SearchResults.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  } else {
+    return (
+      <>
+        <h2>Search results</h2>
+        <div className="search-result-wrapper">
+          {results.map((result) => (
+            <img
+              className="card-image"
+              src={result}
+              alt="searchResultSpaceImage"
```
Comment (StuartCowley) [inline src/components/SearchResults.js:16]: Alt tags are for screen readers, they should be written in plain english

File: src/requests/getImages.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    return axios
+      .get(`https://images-api.nasa.gov/search?q=${query}`)
+      .then((res) => {
+        const imageResults = res.data.collection.items;
+        const parsedImages = imageResults.filter(
+          (mediaObject) => mediaObject.data[0].media_type === 'image'
+        );
+        const images = parsedImages.map((image) => image.links[0].href);
+
+        return images;
```
Comment (StuartCowley) [inline src/requests/getImages.js:16]: You could return the map function result directly:
```
        return parsedImages.map((image) => image.links[0].href);
```

File: src/styles/App.css [hunk 5]
```diff
@@ -0,0 +1,29 @@
+.App {
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+}
+
+.logo {
+  margin-top: 20px;
+  height: 108px;
```
Comment (StuartCowley) [inline src/styles/App.css:9]: You dont need to declare the height value here, as in this case the image will scale automatically based off the width property alone. You can safely delete this

File: src/styles/App.css [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+  color: white;
+}
+
+@media only screen and (max-width: 615px) {
+  .main-title {
+    font-size: 10vw;
+  }
+
+  .logo {
+    width: 30vw;
```
Comment (StuartCowley) [inline src/styles/App.css:26]: Since youre using px outside of the media query, for consistency Id use the same units here

File: src/styles/App.css [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+.main-title {
+  margin-top: 30px;
+  font-size: 80px;
+  color: white;
+}
+
+@media only screen and (max-width: 615px) {
+  .main-title {
+    font-size: 10vw;
```
Comment (StuartCowley) [inline src/styles/App.css:22]: Generally speaking its not a good idea to use vh/vw as units for things like fonts and/or spacing, as they are primarily concerned with larger elements like `div`s. Also, it can cause issues with the interactions between components that are nested inside each particularly when they get to be quite full and complex. 
I would always prefer using `rem` for font sizes and it makes them much easier to manage

File: src/styles/App.css [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+  }
+
+  .logo {
+    width: 30vw;
+    height: auto;
```
Comment (StuartCowley) [inline src/styles/App.css:27]: This is not doing anything and can be taken out

File: src/styles/Search-results.css [hunk 9]
```diff
@@ -0,0 +1,46 @@
+.search-result-wrapper {
+  color: white;
+  display: flex;
+  flex-wrap: wrap;
+  flex-direction: row;
```
Comment (StuartCowley) [inline src/styles/Search-results.css:5]: `flex-direction` is set to `row` implicitly, you dont need to declare it like this unless you are overwriting a previous definition. Can delete this no problem

File: src/styles/Search-results.css [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+
+h2 {
+  color: white;
+  align-self: center;
+}
+
+.card-image {
+  max-width: 100%;
+  max-height: 200px;
+  padding: 1px;
```
Comment (StuartCowley) [inline src/styles/Search-results.css:17]: Is this actually needed?

File: src/styles/Search-results.css [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+  margin-left: 20px;
+}
+
+h2 {
+  color: white;
+  align-self: center;
+}
+
+.card-image {
+  max-width: 100%;
```
Comment (StuartCowley) [inline src/styles/Search-results.css:15]: This is implied so you dont need to specify this

File: src/styles/Search-results.css [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+  max-height: 200px;
+  padding: 1px;
+  border-radius: 5px;
+  overflow: none;
+  align-self: center;
```
Comment (StuartCowley) [inline src/styles/Search-results.css:20]: These properties dont work with this value, they should be taken out

File: src/styles/Search-results.css [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+
+.invalid-search {
+  color: white;
+  align-self: center;
+  margin-top: 40px;
+}
+
+@media only screen and (max-width: 615px) {
+  h2 {
+    font-size: 6vw;
```
Comment (StuartCowley) [inline src/styles/Search-results.css:36]: See point above about using rem for font size

File: src/styles/Search.css [hunk 14]
```diff
@@ -0,0 +1,53 @@
+form {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  flex-wrap: wrap;
```
Comment (StuartCowley) [inline src/styles/Search.css:5]: All these rules can be removed, they arent doing anything

File: src/styles/Search.css [hunk 15]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+.searchBar {
+  align-self: center;
+  margin-left: auto;
+  margin-top: 20px;
+  margin-bottom: 20px;
```
Comment (StuartCowley) [inline src/styles/Search.css:12]: These arent doing anything and can be removed

File: src/styles/Search.css [hunk 16]
```diff
@@ -0,0 +1,53 @@
...
+  border-bottom-width: 3px;
+  border-top-width: 1px;
+  border-left-width: 1px;
+  border-right-width: 1px;
+  border-color: midnightblue;
+  font-family: 'Audiowide';
+}
+
+.searchButton {
+  margin-right: auto;
```
Comment (StuartCowley) [inline src/styles/Search.css:26]: This can be removed

File: src/styles/Search.css [hunk 17]
```diff
@@ -0,0 +1,53 @@
...
+  margin-left: 10px;
+  margin-top: 20px;
+  margin-bottom: 20px;
+  padding: 0 10px;
+  font-size: 20px;
+  border-radius: 30px;
+  align-self: center;
```
Comment (StuartCowley) [inline src/styles/Search.css:33]: This can also be removed

File: src/styles/Search.css [hunk 18]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+@media only screen and (max-width: 615px) {
+  form {
+    flex-direction: column;
+    justify-content: center;
+    align-items: center;
+  }
+  .searchBar {
+    margin: auto;
```
Comment (StuartCowley) [inline src/styles/Search.css:44]: The margin property does not work with the auto keyword for vertical directions, what youre effectively saying here is
```
margin-top: auto;
margin-right: auto;
margin-bottom: auto;
margin-right: auto;
```
This should be changed for 
```
margin: 0 auto;
```

File: src/styles/Search.css [hunk 19]
```diff
@@ -0,0 +1,53 @@
...
+    width: 30vw;
+    font-size: 3vw;
+  }
+  .searchButton {
+    margin: auto;
+    width: 12vw;
+    font-size: 2vw;
```
Comment (StuartCowley) [inline src/styles/Search.css:51]: As above with the font units and margin property

File: src/styles/index.css [hunk 20]
```diff
@@ -0,0 +1,16 @@
...
+    'Helvetica Neue', sans-serif;
+  -webkit-font-smoothing: antialiased;
+  -moz-osx-font-smoothing: grayscale;
+  background-color: rgb(5, 2, 25);
+}
+
+code {
+  font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
+    monospace;
+}
```
Comment (StuartCowley) [inline src/styles/index.css:16]: If youre not using this class anywhere it should be removed

File: README.md [hunk 21]
```diff
@@ -1,70 +1,44 @@
-# Getting Started with Create React App
+# React-Technical-Test (Nasa Space Image Search)
```
Comment (StuartCowley) [inline README.md:1]: Nice readme!

File: src/__tests__/SearchResults.test.js [hunk 22]
```diff
@@ -0,0 +1,10 @@
+import { render } from '@testing-library/react';
+import SearchResults from '../components/SearchResults';
+
+describe('Search', () => {
+  it('renders app component correctly', () => {
+    const { asFragment } = render(<SearchResults />);
+
+    expect(asFragment()).toMatchSnapshot();
+  });
```
Comment (StuartCowley) [inline src/__tests__/SearchResults.test.js:9]: This test is erroring as it cant render the component, there needs to be a prop passed in as with the real component. You can pass in a set of mocked results like this:
```
it("renders app component correctly", () => {
    const validProps = {
      searchResults: ["test-url", "test-url-2"],
    };
    const { asFragment } = render(
      <SearchResults results={validProps.searchResults} />
    );

    expect(asFragment()).toMatchSnapshot();
  });
```
This would create a snapshot as you dont have one in the project for this component yet

You could also run another test here passing in an empty array to verify if the error message would be displayed

File: src/__tests__/App.test.js [hunk 23]
```diff
@@ -0,0 +1,15 @@
+import { render, screen } from '@testing-library/react';
+import App from '../components/App';
+
+describe('App', () => {
+  it('renders app component correctly', () => {
```
Comment (StuartCowley) [inline src/__tests__/App.test.js:5]: The snapshot this creates is out of date, you should always make sure your snapshots are up to date and your tests all pass before submitting for review

**Review by StuartCowley — COMMENTED** [review]
This looks great! There are a few things that could be improved mostly around CSS properties and the tests but all in all theres nothing too great a challenge to sort out. Really good to put your own stamp on the visual aspects too :)
